nilearn/_utils/niimg_conversions.py

```diff
@@ -33,7 +33,54 @@ def _check_same_fov(img1, img2):
             and np.allclose(img1.get_affine(), img2.get_affine()))
 
 
-def check_niimg(niimg, atleast_4d=False):
+def _index_img(img, index):
+    """Helper function for check_niimg_4d."""
+    return new_img_like(
+        img, img.get_data()[:, :, :, index], img.get_affine(),
+        copy_header=True)
+
+
+def _iter_check_niimg(niimgs, ndim=None, atleast_4d=False,
+                      target_fov=None,
+                      memory=Memory(cachedir=None),
+                      memory_level=0, verbose=0):
+    ref_fov = None
+    ndim_minus_one = ndim - 1 if ndim is not None else None
+    if target_fov is not None and target_fov != "first":
+        ref_fov = target_fov
+    for i, niimg in enumerate(niimgs):
```

Comment:
Then I guess niimg should not be used. Maybe `_iter_check_imgs(imgs, ...)` would be better. A short docstring would be great as well.